script.js

```diff
@@ -1,4 +1,206 @@
+var scissors = `scissors`;
+var paper = `paper`;
+var stone = `stone`;
+var reversedScissors = "reversed scissors";
+var reversedPaper = "reversed paper";
+var reversedStone = "reversed stone";
+var replayMessage = `Now you can type "scissors" "paper" or "stone" to play another round!`;
+var numPlayerWins = 0;
+var numComputerWins = 0;
+var numDraw = 0;
+var playerName = ``;
+
+// Generate a random integer from 1 to max inclusive
+var generateRandomInteger = function (max) {
+  var randomInteger = Math.floor(Math.random() * max);
+  return randomInteger;
+};
+
+// // Generate a random roshambo based on random integer
+var generateRandomRoshambo = function () {
+  var numRoshambo = 3;
+  var randomRoshamboNum = generateRandomInteger(numRoshambo);
+  if (randomRoshamboNum == 0) {
+    return scissors;
+  }
+  if (randomRoshamboNum == 1) {
+    return paper;
+  }
+  if (randomRoshamboNum == 2) {
+    return stone;
+  }
+  // If we reach this line, randomRoshamboNum was invalid.
+  return "Oops! There was a bug!";
+};
+
+var getIcon = function (roshambo) {
+  if (roshambo == scissors || roshambo == reversedScissors) {
+    return " ✂️";
+  }
+  if (roshambo == paper || roshambo == reversedPaper) {
+    return " 🗒️";
+  }
+  if (roshambo == stone || roshambo == reversedStone) {
+    return " ✊🏻";
+  }
+};
+
+var getGenericMessage = function (playerRoshambo, computerRoshambo) {
+  var playerRoshamboIcon = getIcon(playerRoshambo);
+  var computerRoshamboIcon = getIcon(computerRoshambo);
+  var genericMessage = `You chose ${playerRoshambo}${playerRoshamboIcon}. <br>The computer chose ${computerRoshambo}${computerRoshamboIcon}.`;
+  return genericMessage;
+};
+
+var getWinsMessage = function (playerName) {
+  var winsMessage = `${playerName} Wins: ${numPlayerWins}, Computer Wins: ${numComputerWins} & Draw: ${numDraw} <br><br>${replayMessage}`;
+  return winsMessage;
+};
+
+/* var validateUser = function (userInput){
+  if ()
+}; */
+
+var validateRoshamboInput = function (roshamboInput) {
+  return (
+    roshamboInput == scissors ||
+    roshamboInput == paper ||
+    roshamboInput == stone ||
+    roshamboInput == reversedScissors ||
+    roshamboInput == reversedPaper ||
+    roshamboInput == reversedStone
+  );
+};
+
```

Comment:
You may have already changed this since we discussed using toLowerCase() as per Sherman's code in class today.

Coding this way => program will not accept reversedscissors or Reversedscissors or any other upper/ lower case combinations but will only accept reversedScissors ... same for reversedPaper and reversedStone